### PR TITLE
feat(LineSegment2D): Add is_equivalent method to line2d and tests.

### PR DESCRIPTION
**Pull request by saeranv**
@chriswmackey 

I'd like to add this is_equivalent method to the LineSegment2D object. Note that order of the points doesn't matter for equivalence to be True.

### CODE REVIEW

**Comment by saeranv** [comment]
@chriswmackey 

I need this for updates to some revised equivalence checks in the ladybug-geometry-polyskel repo, but I admit I don't understand if or how the semantic release tag effects that. 

As long as I do another `pip install ladybug-geometry -U` once this commit is pushed. Shouldn't these changes therefore be correctly propagated to all my other dependent repos (in this case the ladybug-geometry-polyskel one), regardless of the tag?

**Comment by chriswmackey** [comment]
Here's the core library dependency tree:
Ladybug Tools Repo Structures (4)

Making a release at the roots without updating the requirements.txt of the higher branches will cause everything above it to be out of sync. So you will get compatibility warnings when running pip commands like the one you cite there. To avoid this case, our CI makes a new release of everything above a given library whenever something is merged to in order to ensure there's always a compatible set of all the core libraries on PyPI.

Long story short: do you mind getting together your PR for the polyskel library first before we merge this? I just want to avoid the case that  you find that other changes need to be made to ladybug-geometry while you're editing ladybug-geomtery-polyskel and I really don't want to do an extra release of everything if I can avoid it.

**Comment by saeranv** [comment]
@chriswmackey 

Ah I understand, I already have the code done for the polyskel so I can PR it. I don't think I'll need any more changes to this repo.
